docs/rfcs/0003-ecommerce-models.md

```diff
@@ -0,0 +1,136 @@
+## Title for RFC
+
+ecommerce workflow
+
+#### Abstract
+
+This describes the various states of ecommerce models and where and when we can and cannot
+modify data.
+
+#### Product
+
+A `Product` has a `price`, `description`, and a `GenericForeignKey` `content_object` which can
+refer to any Django model instance. An admin user should be able to edit
+`description` or `price` without affecting any previous purchase, so we need to also store
+this separately when the purchase is made.
```

Comment:
should there be a unique constraint on the content <-> product relationship? I can't think of a reason we'd want more than one 'Product' for a course. Although maybe a single course might have different levels of participation or something like that?

Reply:
As far as I can imagine, there's only one product per course run (we sell course runs not courses), but I know that edX allows for different types of seats for some situations (verified, professional, credit, etc). 

We also sell programs, but I think those are just a bundle of courseruns.

Reply:
I think that will work. It looks like GenericForeignKey can support unique constraints:  https://stackoverflow.com/questions/35228902/django-genericforeignkey-and-unique-together. We can remove the unique constraint if something changes in the future.